storage.go

```diff
@@ -86,13 +92,41 @@ func (s *Storage) createAppend(ctx context.Context, path string, opt pairStorage
 	return o, nil
 }
 
+func (s *Storage) createDir(ctx context.Context, path string, opt pairStorageCreateDir) (o *Object, err error) {
+	rp := s.getAbsPath(path)
+
+	res, err := s.client.CreateFolderV2(&files.CreateFolderArg{
+		Path: rp,
+	})
+	if err != nil && checkError(err, files.CreateFolderErrorPath, files.WriteErrorConflict, files.WriteConflictErrorFolder) {
+		// Omit `path/conflict/folder` (`dir exists` related) error here.
+		err = nil
+	}
+	if err != nil {
+		return
+	}
+
+	o = s.newObject(true)
+	o.Mode = ModeDir
+	if res != nil {
```

Comment:
I thought `res` must be valid?

Reply:
`res` is nil when `err` is `path/conflict/folder` returned by `CreateFolderV2` and the error has been omitted.

Reply:
Well, I got it.

If the folder has been created, we need to do following:

```go
o = s.newObject(false)
o.Mode = ModeDir
o.Path = path
```

Instead of set `o.ID = rp` directly.

And please leave comments for that unclear logic. (I didn't get it for my first reading)

Reply:
OK.